clients/src/main/java/org/apache/kafka/common/config/TopicConfig.java

```diff
@@ -63,9 +63,11 @@
         "flush capabilities as it is more efficient.";
 
     public static final String RETENTION_BYTES_CONFIG = "retention.bytes";
-    public static final String RETENTION_BYTES_DOC = "This configuration controls the maximum size a log can grow " +
-        "to before we will discard old log segments to free up space if we are using the " +
-        "\"delete\" retention policy. By default there is no size limit only a time limit.";
+    public static final String RETENTION_BYTES_DOC = "This configuration controls the maximum size a partition " +
+        "(which consists of log segments) can grow to before we will discard old log segments to free up space if we " +
+        "are using the \"delete\" retention policy. By default there is no size limit only a time limit. " +
+        "Please note, this is calculated as retention.bytes * number of partitions on the given topic" +
+        "for the total amount of disk space to be used.";
```

Comment:
Perhaps the last sentence could be "Since this limit is enforced at the partition level, multiply it by the number of partitions to compute the topic retention in bytes"?

Reply:
Sounds good..Updated the PR.

Reply:
@ijuma Can we merge  this minor doc PR